beacon-chain/blockchain/chain_info_test.go

```diff
@@ -119,6 +119,50 @@ func TestCurrentJustifiedCheckpt_CanRetrieve(t *testing.T) {
 	require.Equal(t, cp.Root, bytesutil.ToBytes32(jp.Root))
 }
 
+func TestFinalizedBlockHash(t *testing.T) {
+	ctx := context.Background()
+	beaconDB := testDB.SetupDB(t)
+	fcs := doublylinkedtree.New()
+	opts := []Option{
+		WithDatabase(beaconDB),
+		WithForkChoiceStore(fcs),
+		WithStateGen(stategen.New(beaconDB, fcs)),
+	}
+	service, err := NewService(ctx, opts...)
+	require.NoError(t, err)
+
+	jroot := [32]byte{'j'}
+	cp := &forkchoicetypes.Checkpoint{Epoch: 6, Root: jroot}
+	bState, _ := util.DeterministicGenesisState(t, 10)
+	require.NoError(t, beaconDB.SaveState(ctx, bState, jroot))
+
+	require.NoError(t, fcs.UpdateFinalizedCheckpoint(cp))
+	jp := service.FinalizedBlockHash()
+	require.Equal(t, params.BeaconConfig().ZeroHash, jp)
```

Comment:
Why are you setting in this test jroot for justified root if this is not tested at all? I think the easiest thing is to add a check that the justified checkpoint is being picked `require.Equal(t, jroot, fcs.JustifiedCheckpoint().Root)` or even better use `prepareForkChoiceState()` to set up the node with root `jroot` so that it has the right payload hash. Same applies to other tests in this suite.

Reply:
Gotcha. I'll add the check `require.Equal(t, jroot, fcs.JustifiedCheckpoint().Root)`